open-craft/xblock-group-project-v2
# Validating uploaded file type and extension via django-upload-validator

**Descriptinon:** Adds file upload mime-type and extension validation

**Merge deadline:** Friday June 9th EOD.

**Author concerns:** Due to extremely tight deadline, a couple of corners were cut:

* No unit and integration tests were added.
* List of allowed mime-types and extensions is hard coded, the intent is to make it configurable via XBlock settings.

**Testing instructions:**

0. Set up and configure test environment.
1. Log in as student.
2. Open Group Work
3. In Project Navigator open "Submissions" panel.
4. Try uploading files with different mime-types and extensions
4.1. Allowed extensions: "png", "jpg", "tif", "tiff", "doc", "docx", "xls", "xlsx", "ppt", "pptx", "pdf" - these files should pass validation, but note mime-type check is performed first.
4.2. Everything else should be discarded with 400 error and message stating the mime-type/extension of the file and list of allowed mime-type/extensions.

**Merge checlkist:**

- [x] ~~Documentation updated~~ No updates needed

**After merge:**

- [ ] Bump GPv2 version in edx-platform
